<launch>
    <!-- BSC Mux -->
    <node name="topic_tools_mux" pkg="topic_tools" type="mux"
          args="/cmd_vel /jackal_bsc/bsc_vel /jackal_bsc/key_vel /jackal_bsc/nav_vel mux:=bsc_mux">
    </node>

    <!-- Kinect -->
    <include file="$(find jackal_bsc)/launch/kinect.launch"/>

    <!-- Fake laser scan-->
    <arg name="scan_topic" default="/jackal_laser_scan"/>
    <node name="depthimage_to_laserscan" pkg="depthimage_to_laserscan" type="depthimage_to_laserscan">
        <param name="output_frame_id" value="kinect2_link"/>
        <remap from="image" to="/kinect2/hd/image_depth_rect"/>
        <remap from="camera_info" to="/kinect2/hd/camera_info"/>
        <remap from="scan" to="$(arg scan_topic)"/>
    </node>

    <!-- Move Base -->

    <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">

        <rosparam file="$(find jackal_bsc)/params/costmap_common_params.yaml" command="load" ns="global_costmap" />
        <rosparam file="$(find jackal_bsc)/params/costmap_common_params.yaml" command="load" ns="local_costmap" />

        <rosparam file="$(find jackal_bsc)/params/odom_nav_params/global_costmap_params.yaml" command="load" />
        <rosparam file="$(find jackal_bsc)/params/odom_nav_params/local_costmap_params.yaml" command="load" />

        <rosparam file="$(find jackal_bsc)/params/base_local_planner_params.yaml" command="load" />
        <rosparam file="$(find jackal_bsc)/params/move_base_params.yaml" command="load" />

        <param name="base_global_planner" type="string" value="navfn/NavfnROS" />
        <param name="base_local_planner" value="base_local_planner/TrajectoryPlannerROS"/>

        <remap from="odom" to="odometry/filtered" />
        <remap from="cmd_vel" to="/jackal_bsc/nav_vel"/>
    </node>

</launch>